### PR TITLE
2nd Chunk of Save Data Documentation

### PR DESCRIPTION
**Pull request by ravepossum**
More documentation for savedata.c.

### CODE REVIEW

File: src/savedata.c [hunk 1]
```diff
@@ -579,274 +578,269 @@ static void SaveDataExtra_LoadCheck (SaveData *saveData, int *frontierResult, in
...
 
-    GF_ASSERT(param0 < gSaveTableSize);
-    v0 = v1[param0].sizeFunc();
+    GF_ASSERT(saveTableID < gSaveTableSize);
+    size = saveTable[saveTableID].sizeFunc();
 
-    v0 += 4 - (v0 % 4);
-    v0 += 4;
+    size += 4 - (size % 4);
+    size += 4;
```
Comment (lhearachel) [inline src/savedata.c:841]: Is there a potential const for this 4?

File: include/savedata.h [hunk 2]
```diff
@@ -33,17 +33,17 @@ typedef struct SaveDataBody {
...
-    u32 unk_1C[2];
-    volatile BOOL unk_24;
-} UnkStruct_020250DC;
+typedef struct SaveDataState {
+    BOOL fullSaveMode;
+    int startBlock;
+    int currentBlock;
+    int endBlock;
+    s32 lockID;
+    int stateSequence;
```
[the remark this answers is not in the document]
Comment (ravepossum) [reply include/savedata.h]: mainSequence could be a bit more appropriate there.

File: src/savedata.c (hunk 1 above)
Comment (ravepossum) [reply src/savedata.c:843]: From what I can tell it looks like padding, so the significance just seems to be that it's half a byte. I'll plan on leaving as-is unless you disagree.

Comment (lhearachel) [reply src/savedata.c:843]: Fine by me.

**Review by lhearachel — APPROVED** [review]

**Pull request merged**